db.go

```diff
@@ -562,6 +562,20 @@ func (db *DB) Select(dest interface{}, q interface{}, args ...interface{}) error
 	return selectObjects(db, dest, q, args)
 }
 
+// SelectOne runs an arbitrary SQL query, unmarshalling the matching row
+// into the the struct specified by dest. Args are the
+// parameters to the SQL query. The query is expected to return at most one row.
+//
+// It is ok for dest to refer to a struct that has not been bound to a
+// table. This allows querying for values that return transient
+// columnts. For example, "SELECT count(*) ..." will return a "count"
+// column.
+//
+// dest must be a pointer to a struct.
+func (db *DB) SelectOne(dest interface{}, query interface{}, args ...interface{}) error {
+	return getObjectByQuery(db, db, dest, query, args...)
+}
+
```

Comment:
I think similar functionality can be had from `QueryRow(...).Scan(&val)`. I'm not sure it is worthwhile to support both methods and the `QueryRow` version has the advantage of matching up with the `database/sql` interfaces.